### PR TITLE
Fix phi node value renumbering issue

### LINKED ISSUE #28077: Miscompile for `@goto`
```julia
function foo()
    i = 0
    @label L
    i += 1
    println(i)
    if i < 10
        @goto L
    end
    return i
end
```
Calling `foo()` keeps printing 1 forever. (On 0.6, this prints the numbers 1 to 10 as expected.) Indeed:
```julia
julia> @code_typed foo()
CodeInfo(
  1 ─      nothing                                                                                                         │ 
4 2 ┄ %2 = φ (1 => 0, 3 => %2)::Int64                                                                                      │ 
  │   %3 = Base.add_int(%2, 1)::Int64                                                                                      │╻ +
5 │        invoke Main.println(%3::Int64)                                                                                  │ 
6 │   %5 = Base.slt_int(%3, 10)::Bool                                                                                      │╻ <
  └──      goto 4 if not %5                                                                                                │ 
7 3 ─      goto 2                                                                                                          │ 
9 4 ─      return %3                                                                                                       │ 
) => Int64
```
I'd say the φ-node should be
```
4 2 ┄ %2 = φ (1 => 0, 3 => %3)::Int64                                                                                      │ 
```
I haven't bisected, but the new optimizer would be the usual suspect, CC @Keno

### PR DESCRIPTION
**Pull request by martinholters**
If a φ node got placed at a position one of its values referenced, that reference wasn't updated.
Fixes #28077.

### CODE REVIEW

File: base/compiler/ssair/ir.jl [hunk 1]
```diff
@@ -679,7 +679,7 @@ function process_phinode_values(old_values::Vector{Any}, late_fixup::Vector{Int}
                 used_ssas[val.id] += 1
             end
         elseif isa(val, OldSSAValue)
-            if val.id > processed_idx
+            if val.id >= processed_idx
```
Comment (martinholters) [inline base/compiler/ssair/ir.jl]: The test passes without this one. I do think the change is correct, but if someone can come up with a piece of code that comes here with `val.id==processed_idx` to build a test case, that would be appreciated.

**Comment by martinholters** [comment]
freebsd failure is #28101. 
Circle says: "Your current usage represents 78% of your JuliaLang Linux plan's limit." Looking at the queue, that percentage sounds, eh, slightly underestimated.

**Comment by Keno** [comment]
Actually, that's still incorrect. I think `idx-1` is always correct. I'll update the PR.

**Review by Keno — APPROVED** [review]

**Comment by martinholters** [comment]
The fact that we had three similar, but not equivalent patches that all fixed the issue and did not break any other tests makes me a bit uneasy. Ideally, we would have tests that had failed for the wrong fixes here. I'm clueless how to construct such tests, though.

**Pull request merged**